### DISCUSSION TITLE
Cannot Use Custom Shapes

### DISCUSSION THREAD

**Discussion by ahmutkadim** [replit/kaboom#846] [Q&A] [answered]
Hi,

I am very new to Kaboom.js and have problem with adding custom shape to an area component. Here is the code snippet for object creation:

```js
let player = k.add([
	k.sprite("player"),
	k.area({shape: k.rect(20, 30)}),
	k.pos(320, 180),
	k.body({jumpForce: 300}),
]);
```

The error it gives: "Only support polygon and rect shapes for now"

Am I doing something wrong or is this a bug?

**Comment by lajbel** [comment, marked as answer]
no documented, sorry.

You should use 

```js
k.area({
    shape: new k.Rect(k.pos(0), 20, 30),
})
```
